lib/html_parser.dart

```diff
@@ -342,7 +342,8 @@ class HtmlParser extends StatelessWidget {
     }
 
     //Return the correct InlineSpan based on the element type.
-    if (tree.style.display == Display.BLOCK && tree.children.isNotEmpty) {
+    if (tree.style.display == Display.BLOCK &&
+        (tree.children.isNotEmpty || tree.element?.localName == "hr")) {
```

Comment:
Note to self: I don't think there are any other tags that render as block yet have no children. cf https://developer.mozilla.org/en-US/docs/Glossary/Empty_element